pkg/logs/internal/launchers/file/launcher.go

```diff
@@ -342,7 +342,7 @@ func (s *Launcher) createRotatedTailer(t *tailer.Tailer, file *tailer.File, patt
 }
 
 func CheckProcessTelemetry(stats *util.ProcessFileStats) {
-	if stats.AgentOpenFiles/stats.OsFileLimit > 0.9 {
+	if float64(stats.AgentOpenFiles/stats.OsFileLimit) > 0.9 {
```

Comment:
This will be true only if `AgentOpenFiles >= OsFileLimit`, because division happens while the values are still integers: https://go.dev/play/p/PbT1MxAia1o

It may not be practical to write a test for this behavior, but please include a QA step to verify that this still works.

Reply:
Applied `float64(x)` to each number individually to address this. Tested with:
```
	if float64(stats.AgentOpenFiles)/float64(stats.OsFileLimit)+.9 > 0.9 {
		log.Warnf("Agent process is close to OS file limit of %v. Agent process currently has %v files open. Ratio is %v.", stats.OsFileLimit, stats.AgentOpenFiles, float64(stats.AgentOpenFiles)/float64(stats.OsFileLimit))
```
Output below:
```
2023-03-13 18:05:31 UTC | CORE | WARN | (pkg/logs/internal/launchers/file/launcher.go:346 in CheckProcessTelemetry) | Agent process is close to OS file limit of 1048576. Agent process currently has 20 files open. Ratio is 1.9073486328125e-05.
```